consensus/hotstuff/signature/weighted_signature_aggregator_test.go

```diff
@@ -238,4 +242,174 @@ func TestWeightedSignatureAggregator(t *testing.T) {
 		assert.Nil(t, signers)
 	})
 
+	//  2. A signature added via `TrustedAdd` has an invalid structure (fails to deserialize)
+	//      2.a. aggregated public key is not identity
+	//      2.b. aggregated public key is identity
+	t.Run("invalid signature serialization", func(t *testing.T) {
+		ids, pks, sigs, msg, _, tag := createAggregationData(t, signersNum)
+		// sigs[0] has an invalid struct
+		sigs[0] = (crypto.Signature)([]byte{0, 0})
+
+		t.Run("with non-identity aggregated public key", func(t *testing.T) {
+			aggregator, err := NewWeightedSignatureAggregator(ids, pks, msg, tag)
+			require.NoError(t, err)
+
+			// test Verify
+			err = aggregator.Verify(ids[0].NodeID, sigs[0])
+			assert.ErrorIs(t, err, model.ErrInvalidSignature)
+
+			// add signatures for aggregation including corrupt sigs[0]
+			expectedWeight := uint64(0)
+			for i, sig := range sigs {
+				weight, err := aggregator.TrustedAdd(ids[i].NodeID, sig)
+				require.NoError(t, err)
+				expectedWeight += ids[i].Weight
+				assert.Equal(t, expectedWeight, weight)
+			}
+
+			// Aggregation should error with sentinel InvalidSignatureIncludedError
+			// aggregated public key is not identity (equal to sum of all pks)
+			signers, agg, err := aggregator.Aggregate()
+			assert.True(t, model.IsInvalidSignatureIncludedError(err))
+			assert.Nil(t, agg)
+			assert.Nil(t, signers)
+		})
+
+		t.Run("with identity aggregated public key", func(t *testing.T) {
+			// assign  pk1 to -pk0 so that the aggregated public key is identity
+			pkBytes := pks[0].Encode()
+			negatePoint(pkBytes)
+			var err error
+			pks[1], err = crypto.DecodePublicKey(crypto.BLSBLS12381, pkBytes)
+			require.NoError(t, err)
+
+			// aggregator with two signers
+			aggregator, err := NewWeightedSignatureAggregator(ids[:2], pks[:2], msg, tag)
+			require.NoError(t, err)
+
+			// add the invalid signature on index 0
+			_, err = aggregator.TrustedAdd(ids[0].NodeID, sigs[0])
+			require.NoError(t, err)
+
+			// add a second signature for index 1
+			_, err = aggregator.TrustedAdd(ids[1].NodeID, sigs[1])
+			require.NoError(t, err)
+
+			// Aggregation should error with sentinel InvalidSignatureIncludedError
+			// aggregated public key is identity
+			signers, agg, err := aggregator.Aggregate()
+			assert.True(t, model.IsInvalidSignatureIncludedError(err))
```

Comment:
I think we could allow here to return either `InvalidSignatureIncludedError` or  `InvalidAggregatedKeyError`